src/charm.py

```diff
@@ -207,6 +219,27 @@ def _on_restart(self, event):
         event.set_results({"result": "ranger successfully restarted"})
         self.unit.status = ActiveStatus()
 
+    def set_truststore_password(self, container):
+        """Update the truststore password to the randomly generated one.
+
+        Args:
+            container: The application container.
+        """
+        command = [
+            f"{JAVA_HOME}/bin/keytool",
+            "-storepass",
+            "changeit",
+            "-storepasswd",
+            "-new",
+            self._state.truststore_pwd,
+            "-keystore",
+            f"{JAVA_HOME}/lib/security/cacerts",
+        ]
+        try:
+            container.exec(command).wait()
+        except ExecError as e:
+            logger.debug(f"Truststore password is already updated: {e.stderr}")
```

Comment:
question: How can you be sure that is the error reason?

Reply:
Good point, I've added a string check for the error.